src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -120,6 +121,9 @@ public function getBuildDir(): string
         if (isset($_SERVER['APP_BUILD_DIR'])) {
             return $_SERVER['APP_BUILD_DIR'].'/'.$this->environment;
         }
+        if ('prod' === $this->getEnvironment() && Kernel::class === (new \ReflectionMethod(parent::class, 'getCacheDir'))->class) {
+            return $this->getProjectDir().'/var/build/'.$this->environment;
+        }
```

Comment:
it feels a bit dirty ... what about a new (temporary?) configuration env? to be used upstream https://github.com/symfony/symfony/blob/3729c5c7d74fd87ee76581a1fb8b97e9db7cb2ef/src/Symfony/Component/HttpKernel/Kernel.php#L302-L303

which brings me to; can we get rid of the parent calls in a trait, by making the envs defacto config envs

Reply:
I don't understand your proposal sorry (nor its "why"). Please expand if you want.

Reply:
e.g. `APP_USE_BUILD_DIR=1` so we dont hardcode the prod env and rely on reflection

also im wondering handling the APP_ envs like APP_CACHE_DIR can be dealt with in `Kernel` as a firt clas feature

Reply:
An alternative implementation might be to define APP_BUILD_DIR in .env.prod in a recipe.

Reply:
:+1: i tend to prefer a pure configuration solution

but isnt the bigger goal to eventually swap the current implementation?
https://github.com/symfony/symfony/blob/3729c5c7d74fd87ee76581a1fb8b97e9db7cb2ef/src/Symfony/Component/HttpKernel/Kernel.php#L300-L304